forge/forge.go

```diff
@@ -175,6 +176,20 @@ func (c *Client) NewRequest(ctx context.Context, method string, urlStr string, b
 	return req, nil
 }
 
+// checkResponseError checks if response is an error.
+func checkResponseError(res *http.Response) error {
```

Comment:
I wonder if we should marshal the response sent back when an error occurs?

It could easily be modelled like..

```
type ForgeError struct {
  Message string
  Errors []string
}
```

Discussion point for later maybe